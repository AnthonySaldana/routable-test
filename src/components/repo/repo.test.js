import React from 'react';
import ReactDOM from 'react-dom';
import Repo from '../repo';

const mockProps = {
    repo: {
        name: "Test Repo",
        open_issues: 5
    }
};

let elem;

beforeEach(() => {
    elem = document.createElement('div');
    ReactDOM.render(<Repo repo={mockProps.repo} />, elem);
});

it('Matches Snapshot', () => {
  expect(elem).toMatchSnapshot();
});

it('Repo Name matches up', () => {
    expect(elem.getElementsByClassName('repoName')[0].innerHTML).toEqual('Test Repo');
});

it('Repo Open Issues matches up', () => {
    expect(elem.getElementsByClassName('repoIssues')[0].innerHTML).toEqual('Open Issues: 5');
});