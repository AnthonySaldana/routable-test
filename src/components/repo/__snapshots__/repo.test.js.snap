// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div>
  <div
    class="repoCard"
  >
    <p
      class="repoName"
    >
      Test Repo
    </p>
    <p
      class="repoIssues"
    >
      Open Issues: 
      5
    </p>
  </div>
</div>
`;
